File: frontend/src/types/offer.ts
export interface OfferTemplate {
  id: string;
  name: string;
  department: string | null;
  role_type: string | null;
  content: string;
  placeholders: string[] | null;
  is_active: number;
  created_at: string;
  updated_at: string;
}

export type TemplateCreate = Pick<OfferTemplate, 'name' | 'content'> & {
  department?: string;
  role_type?: string;
  placeholders?: string[];
};

export type TemplateUpdate = Partial<TemplateCreate & { is_active: number }>;

export interface Offer {
  id: string;
  job_id: string;
  candidate_id: string;
  template_id: string | null;
  content: string;
  offer_data: Record<string, unknown> | null;
  pdf_path: string | null;
  status: string;
  generation_model: string | null;
  created_at: string;
  updated_at: string;
}

export interface OfferGenerateRequest {
  job_id: string;
  candidate_id: string;
  template_id: string;
  offer_data: Record<string, unknown>;
}

File: backend/app/schemas/offer.py
from datetime import datetime

from pydantic import BaseModel


class TemplateCreate(BaseModel):
    name: str
    department: str | None = None
    role_type: str | None = None
    content: str
    placeholders: list[str] | None = None


class TemplateUpdate(BaseModel):
    name: str | None = None
    department: str | None = None
    role_type: str | None = None
    content: str | None = None
    placeholders: list[str] | None = None
    is_active: int | None = None


class TemplateResponse(BaseModel):
    id: str
    name: str
    department: str | None
    role_type: str | None
    content: str
    placeholders: list[str] | None
    is_active: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class OfferGenerateRequest(BaseModel):
    job_id: str
    candidate_id: str
    template_id: str
    offer_data: dict  # salary, start_date, benefits, etc.


class OfferUpdate(BaseModel):
    content: str | None = None
    offer_data: dict | None = None


class OfferResponse(BaseModel):
    id: str
    job_id: str
    candidate_id: str
    template_id: str | None
    content: str
    offer_data: dict | None
    pdf_path: str | None
    status: str
    generation_model: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
